rrmngmnt/host.py

```diff
@@ -94,12 +94,14 @@ def add(self):
         """
         Add host to inventory
         """
-        try:
-            host = self.get(self.ip)
-        except ValueError:
-            pass
-        else:
-            self.inventory.remove(host)
+        lock = threading.Lock()
```

Comment:
it doesn't work like this, you can not create instance of lock for every call

you should add it as classmember, and use it here